test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java

```diff
@@ -68,45 +70,66 @@
 
     private static final boolean USE_THROW_CATCH = false; // Test bugs
 
-    private static final Optional<MemoryPoolMXBean> NON_SEGMENTED_CODE_CACHE_POOL;
-    private static final Optional<MemoryPoolMXBean> NON_NMETHODS_POOL;
-    private static final Optional<MemoryPoolMXBean> PROFILED_NMETHODS_POOL;
-    private static final Optional<MemoryPoolMXBean> NON_PROFILED_NMETHODS_POOL ;
-
-    // Limit numbers are arbitrary, feel free to change if arguably necessary
-    private static final int NON_SEGMENTED_CACHE_ALLOWANCE = 2_000_000;
-    private static final int SEGMENTED_CACHE_ALLOWANCE = 1_000_000;
-
-    static {
-        var pools = ManagementFactory.getMemoryPoolMXBeans();
-        NON_SEGMENTED_CODE_CACHE_POOL = pools.stream()
-            .filter(pool -> pool.getName().equals("CodeCache")).findFirst();
-        NON_NMETHODS_POOL = pools.stream()
-            .filter(pool -> pool.getName().equals("CodeHeap 'non-nmethods'")).findFirst();
-        PROFILED_NMETHODS_POOL = pools.stream()
-            .filter(pool -> pool.getName().equals("CodeHeap 'profiled nmethods'")).findFirst();
-        NON_PROFILED_NMETHODS_POOL = pools.stream()
-            .filter(pool -> pool.getName().equals("CodeHeap 'non-profiled nmethods'")).findFirst();
-    }
+    /**
+     * The class is used for periodical checks if a code-cache consuming operation
+     * could be executed (i.e. if code cache has enought free space for a typical operation).
+     */
+    private static class CodeCacheMonitor {
+
+        private static final Optional<MemoryPoolMXBean> NON_SEGMENTED_CODE_CACHE_POOL;
+        private static final Optional<MemoryPoolMXBean> NON_NMETHODS_POOL;
+        private static final Optional<MemoryPoolMXBean> PROFILED_NMETHODS_POOL;
+        private static final Optional<MemoryPoolMXBean> NON_PROFILED_NMETHODS_POOL;
+
+        // Trial runs show up that maximal increase in code cache consumption between checks (for one
+        // cycle/tree build in MHTransformationGen::createSequence), falls within the following intervals:
+        //
+        // | Threads number | Without Xcomp | With Xcomp |
+        // |----------------|---------------|------------|
+        // |       1        |   100-200 K   |  400-500 K |
+        // |      10        |    1 - 2 M    |    5-6 M   |
+        //
+        // Those numbers are approximate (since threes are generated randomly and the total consumption
```

Comment:
s/threes/trees/ ?

Reply:
> s/threes/trees/ ?

How did you know that these were not actual threes / troikas? :))

Fixed, thanks.